assembly/src/v1/parsers/io_ops.rs

```diff
@@ -151,6 +202,26 @@ fn push_mem_addr(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Assemb
     Ok(())
 }
 
+// HELPERS
+// ================================================================================================
+
+/// This is a helper function that validates the length of the assembly io instruction and returns
+/// an error if params are missing or there are extras.
+///
+/// The minimum and maximum number of valid instruction parts for the op are provided by the
+/// caller.
+///
+/// # Errors
+///
+/// This function will return an AssemblyError if the assembly op has too many or too few parts.
+fn validate_op_len(op: &Token, min_parts: usize, max_parts: usize) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        too_few if too_few < min_parts => Err(AssemblyError::missing_param(op)),
+        too_many if too_many > max_parts => Err(AssemblyError::extra_param(op)),
+        _ => Ok(()),
+    }
+}
+
```

Comment:
Not for this PR, but I wonder if we should move this out to be used in other modules as well.

Reply:
Yes, I think we should. Otherwise we're either duplicating code or we're under-validating. Pulling this logic out should make all of the op parsing implementations more readable.